libs/core/src/lib/step-input/step-input.component.html

```diff
@@ -0,0 +1,55 @@
+<div class="fd-step-input"
+     [ngClass]="state ? 'is-' + state : ''"
+     [class.is-focus]="focused"
+     [class.is-readonly]="readonly"
+     [class.is-disabled]="disabled"
+     [class.fd-step-input--compact]="compact"
+     [class.fd-step-input--without-buttons]="!hasStepButtons">
+    <button *ngIf="hasStepButtons"
+            fd-button
+            #decrementBtn
+            fdType="transparent"
+            class="fd-step-input__button"
+            tabindex="-1"
+            type="button"
+            [compact]="compact"
+            [disabled]="!canDecrement"
+            [title]="decrementButtonTitle"
+            [glyph]="decrementButtonIcon"
+            (click)="onTouched()">
+    </button>
+    <input #inputElement
+           [name]="name"
+           [id]="inputId"
+           [title]="inputTitle"
+           [value]="viewValue"
+           [disabled]="disabled"
+           [readOnly]="readonly"
+           [class.fd-input--compact]="compact"
+           [attr.aria-label]="ariaLabel"
+           (focusin)="handleFocusIn()"
+           (focusout)="handleFocusOut()"
+           (wheel)="handleScroll($event)"
+           (keydown)="handleKeyDown($event)"
+           (input)="trackInputValue($event)"
+           (change)="updateViewValue($event)"
+           [style.textAlign]="textAlign"
+           class="fd-input fd-input--no-number-spinner fd-step-input__input"
+           type="text">
+    <button *ngIf="hasStepButtons"
+            fd-button
+            #incrementBtn
+            fdType="transparent"
+            class="fd-step-input__button"
+            tabindex="-1"
+            type="button"
+            [compact]="compact"
+            [disabled]="!canIncrement"
+            [title]="incrementButtonTitle"
+            [glyph]="incrementButtonIcon"
+            (click)="onTouched()">
+    </button>
+</div>
+<span class="fd-form-label fd-form-label--unit-description">
```

Comment:
Use ngIf so that it doesnt have padding or an empty label in case user does not add one